static/js/recent_topics_ui.js

```diff
@@ -617,6 +618,11 @@ export function initialize_handler() {
 }
 
 export function show() {
+    const was_narrowed_already = narrow_state.active();
+    if (was_narrowed_already || message_lists.current === message_lists.home) {
+        narrow.save_pre_narrow_offset_for_reload();
+    }
```

Comment:
Is this meant to be just a check for whether we previously had a message feed open?  (I guess the alternative is that we just loaded the app looking at "Recent topics"?).  I feel like there must be a better way to write a check for that, though might require a bit of research to find it.

Reply:
Yes, that's what it's meant to check.
Since we just did the plumbing for `recent_topics_ui.initialize_handler`, it wouldn't be too difficult to add a `is_recent_topics_first_view` variable,  but I'm not certain if it's better or worse.

Reply:
I don't think `is_recent_topics_first_view` makes sense; it should be something from the `narrow.js` state machine.